### DISCUSSION TITLE
Set query language

### DISCUSSION THREAD

**Discussion by traysh** [toverainc/willow#185] [Q&A] [answered]
I'm using my ESP-BOX and my automations only in (Brazilian) Portuguese, and I noticed having the `detect_language=true` parameter AND speaking in Portuguese takes a lot longer than when speaking in English.

 I'd like to specify the input and output language in the query to the inference server. I tried setting the 'return_language=pt', but I'm still receiving responses in English. The url is bellow:

```
https://infer.tovera.io/api/willow?detect_language=false&return_language=pt
```

Shouldn't I be receive responses in Portuguese?

**Comment by kristiankielhofner** [comment, marked as answer]
Language handling has been completely revamped in our next release of WIS. You can test it if you use this inference server URL in the Willow Configuration:

```https://wisng.tovera.io/api/willow?force_language=pt```

**Reply by traysh** [reply]
Great, thank you for the information!

**Reply by kristiankielhofner** [reply]
You're welcome! We are planning on releasing this as WIS 1.0 and we've been specifically looking for non-English speakers to test the new implementation (you're the first with Portuguese). Let us know how it goes!

**Reply by traysh** [reply]
It works great, and is super fast!

**Reply by kristiankielhofner** [reply]
Great, glad to hear!
